xbmc/guilib/guiinfo/SystemGUIInfo.cpp

```diff
@@ -182,11 +182,11 @@ bool CSystemGUIInfo::GetLabel(std::string& value, const CFileItem *item, int con
       return true;
     case SYSTEM_SCREEN_RESOLUTION:
     {
-      RESOLUTION_INFO& resInfo = CDisplaySettings::GetInstance().GetCurrentResolutionInfo();
+      const RESOLUTION_INFO& resInfo = CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo();
```

Comment:
maybe save `CServiceBroker::GetWinSystem()` first since in the if condition you get it again to check the full screen status?

Reply:
Good idea, and because guilib can't assume there's a winsystem, I think better would be:

```
--- a/xbmc/guilib/guiinfo/SystemGUIInfo.cpp
+++ b/xbmc/guilib/guiinfo/SystemGUIInfo.cpp
@@ -182,14 +182,22 @@ bool CSystemGUIInfo::GetLabel(std::string& value, const CFileItem *item, int con
       return true;
     case SYSTEM_SCREEN_RESOLUTION:
     {
-      RESOLUTION_INFO& resInfo = CDisplaySettings::GetInstance().GetCurrentResolutionInfo();
-      if (CServiceBroker::GetWinSystem()->IsFullScreen())
+      CWinSystemBase* winSystem = CServiceBroker::GetWinSystem();
+      if (winSystem != nullptr)
+      {
+        const RESOLUTION_INFO& resInfo = winSystem->GetGfxContext().GetResInfo();
+        if (winSystem->IsFullScreen())
           value =
               StringUtils::Format("{}x{}@{:.2f}Hz - {}", resInfo.iScreenWidth, resInfo.iScreenHeight,
                                   resInfo.fRefreshRate, g_localizeStrings.Get(244));
         else
           value = StringUtils::Format("{}x{} - {}", resInfo.iScreenWidth, resInfo.iScreenHeight,
                                       g_localizeStrings.Get(242));
+      }
+      else
+      {
+        value = "";
+      }
       return true;
     }
     case SYSTEM_BUILD_VERSION_SHORT:
```

Reply:
You can also modify the above patch to do `const CWinSystemBase* winSystem = CServiceBroker::GetWinSystem();` (because the guilabel is a getter) and add a const getter for the CGraphicContext to CWinSystemBase:

```
--- a/xbmc/windowing/WinSystem.cpp
+++ b/xbmc/windowing/WinSystem.cpp
@@ -267,6 +267,11 @@ void CWinSystemBase::DriveRenderLoop()
   }
 }
 
+const CGraphicContext& CWinSystemBase::GetGfxContext() const
+{
+  return *m_gfxContext;
+}
+
 CGraphicContext& CWinSystemBase::GetGfxContext()
 {
   return *m_gfxContext;
diff --git a/xbmc/windowing/WinSystem.h b/xbmc/windowing/WinSystem.h
index cac94b06d6..0e415739a3 100644
--- a/xbmc/windowing/WinSystem.h
+++ b/xbmc/windowing/WinSystem.h
@@ -148,6 +148,7 @@ public:
   virtual bool MessagePump() { return false; }
 
   // Access render system interface
+  const CGraphicContext& GetGfxContext() const;
   CGraphicContext& GetGfxContext();
 
   /**
```